cluster.tf

```diff
@@ -54,11 +54,12 @@ resource "aws_security_group_rule" "cluster_https_worker_ingress" {
 
 resource "aws_iam_role" "cluster" {
   count                 = "${var.manage_cluster_iam_resources ? 1 : 0}"
-  name_prefix           = "${var.cluster_name}"
+  name                  = "${local.iam_role_name}"
```

Comment:
This will cause the role to be recreate, correct? What happens to the cluster when this happens? Is it also recreated?

Reply:
> This will cause the role to be recreate, correct
yes, this is a resource recreate.

> What happens to the cluster when this happens? Is it also recreated?
yeah, we get a `forces new resource` on the cluster too, I think. Would you want me to confirm? i can do so after lunch today.

Reply:
@max-rocket-internet here are the resources that want to be recreated when `aws_iam_role` is changed.

```hcl
-/+ module.eks.aws_eks_cluster.this (new resource required)
-/+ module.eks.aws_iam_role.cluster (new resource required)
-/+ module.eks.aws_iam_role_policy_attachment.cluster_AmazonEKSClusterPolicy (new resource required)
-/+ module.eks.aws_iam_role_policy_attachment.cluster_AmazonEKSServicePolicy (new resource required)
-/+ module.eks.aws_launch_configuration.workers (new resource required)
-/+ module.eks.null_resource.update_config_map_aws_auth (new resource required)
```

Reply:
after reading [some issue comments](https://github.com/terraform-aws-modules/terraform-aws-eks/issues/150#issuecomment-427057836) it seems this is a bigger deal than i expected or consider.

@max-rocket-internet do we want to go ahead and close this out or evolve this more in line with general consensus (based on existing issue read throughs)?

my case is that i use cloud posse's [null-label module](https://github.com/cloudposse/terraform-null-label) and pass `"${module.null-label_this.tags}"` to any resource that supports tagging.

once EKS is up, it has an iam role. it's useful to be able to find said role via `data` resource filtering on tags instead of doing a `remote_state` call (undocumented but i do remember seeing a post where it was silently added during one of the later updates to tf 0.11).

I'll follow the intent [here](https://github.com/terraform-aws-modules/terraform-aws-eks/issues/88#issuecomment-482757743) and add region to the role name suffix instead of random string.

Reply:
> it seems this is a bigger deal than i expected or consider.

Exactly, that's what I'm getting at. We can't merge it if it will recreate the cluster 😅

Reply:
I still don't get why you need to change the name of the role just add tags?

Reply:
Or why we need a random string in the name?

Reply:
> I still don't get why you need to change the name of the role just add tags

we don't need to change the name. that was an artifact for a previous effort here at work. cleaning up now.

Reply:
> why we need a random string in the name?

the intent was to as closely match the result of `name_prefix` - again for something internal here at work. removed.

Reply:
> We can't merge it if it will recreate the cluster 😅

totally. thanks for your patience.